### PR TITLE
Update SafeDNS provider details and error handling

### PR DESCRIPTION
**Pull request by miff2000**
A few changes, mostly to the SafeDNS provider:

* Correct `NAMESERVER_DOMAINS` value.
* Update API token page link
* Improve error handling for 404 Not Found backend API responses, fixing premature end of zone name list iteration in Certbot
* Update domain name used in tests
* Update Cassettes for SafeDNS API
* Correct path to provider test files in `CONTRIBUTING.md`.

### CODE REVIEW

**Comment by miff2000** [comment]
Hey @adferrand, are you able to take a look at this PR for me? Thanks 😊

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -154,7 +154,7 @@ In default test mode, tests are replayed from existing recordings. In live mode,
 
 To execute the `py.test` suite using the live tests mode, execute py.test with the environment variable `LEXICON_LIVE_TESTS` set to `true` like below:
 
-	LEXICON_LIVE_TESTS=true py.test tests/providers/test_foo.py
+	LEXICON_LIVE_TESTS=true py.test lexicon/tests/providers/test_foo.py
```
Comment (adferrand) [inline CONTRIBUTING.md]: Thanks for the fix in the documentation!

File: lexicon/providers/safedns.py [hunk 2]
```diff
@@ -8,14 +8,14 @@
 
 LOGGER = logging.getLogger(__name__)
 
-NAMESERVER_DOMAINS = ['graphiterack.com']
+NAMESERVER_DOMAINS = ['ukfast.net']
```
Comment (adferrand) [inline lexicon/providers/safedns.py]: Are we sure that the previous nameserver domains is not use anymore? Note that we can have several values in the array, the `auto` provider (that consumes the `NAMESERVER_DOMAINS` values in other providers), is able to handle this situation, and will check against each domain provided here.

File: CONTRIBUTING.md (hunk 1 above)
Comment (miff2000) [reply CONTRIBUTING.md]: Any time! 😊

File: lexicon/providers/safedns.py (hunk 2 above)
Comment (miff2000) [reply lexicon/providers/safedns.py]: The `graphiterack.com` domain is used for our internal recursive nameservers, not the authoritative ones, so I thought it best to leave them off. Any the nameserver entries for customers that we cost domains for use `ns0.ukfast.co.uk` and `ns1.ukfast.co.uk`

**Pull request merged**